### DISCUSSION TITLE
Input not working in the final build from the project export

### DISCUSSION THREAD

**Discussion by alhmoraansel** [ezEngine/ezEngine#1245] [Q&A]
I just downloaded the release Dec2023 and opened the Testing chambers project... explored options and finally tried exporting the project... However when i run the exported project, input does not works

https://github.com/ezEngine/ezEngine/assets/109255259/324b21dd-d57f-4e91-af5c-3a12a20a8ea3

**Comment by jankrassnigg** [comment]
Can't test this right now, since I'm out over the Easter holidays. There is a bug that sometimes makes typescript scripts not work.
As far as I remember, deleting the asset cache should fix this. There is an option in the editor to do that in the "project" menu.
After that make sure to transform all assets again.
Sorry can't say more atm.

**Reply by alhmoraansel** [reply]
Thanks for replying... I'll try doing that.

**Comment by AlexandrKedrov** [comment]
Hi! It looks like AngelScript have similar issue, however, I couldn't make it work even after deleting AssetCache folder.
Also RML UI interface appears blank after the project export 
<img width="1282" height="752" alt="image" src="https://github.com/user-attachments/assets/cb37518d-14bf-4364-9da4-9fbca458b964" />

**Reply by jankrassnigg** [reply]
Haven't seen this before. After deleting the asset caches ("Project > Clear Asset Cache > No to all") make sure to transform everything. E.g. go to your main scene, Ctrl+E and check "Transform all". Afterwards also check the Asset Curator panel, that there are no issues somewhere.

If it persists even then, I'd need a repro case, to test with.

**Reply by AlexandrKedrov** [reply]
Thanks for the reply. 

I tried the suggested scheme with a clean dev build and with different projects, but the issue still persists. 

The AngelScript issue can be reproduced in Testing Chambers. Also I have attached a minimal test project that contains only a simplified Player prefab from TC and it reproduces the issue as well

[TestExport.zip](https://github.com/user-attachments/files/28477649/TestExport.zip)

The issue with blank RML UI elements can be reproduced in the RTS Sample 
<img width="1913" height="1080" alt="image" src="https://github.com/user-attachments/assets/84672b3d-e4b2-42d3-ac43-2d4182be6f2c" />

**Reply by jankrassnigg** [reply]
I can't reproduce this. Not with the RTS sample, not with your project. Tried a Debug and a Dev build. I've also not encountered this issue in general. The input problem that I do know about, was specific to TypeScript and is gone since we switched to AngelScript.

This appears to be specific to your machine. Maybe some anit-virus software that's blocking access to certain files?

I think you'll have to debug this issue. Check the logs for any errors. Put a breakpoint in `ezAngelScriptResource::UpdateContent` to see whether any resource fails to load. The RML UI stuff seems to load for you at least partially, not sure where to check what's going wrong there, though it should show up in the log.

Build the engine in a Debug build, and use the plugin that's mentioned here (https://ezengine.net/pages/docs/debugging/debug-cpp.html) to attach to the child processes.

Sorry, I don't think I can help you more than that directly, but let me know if you have further questions.

**Reply by AlexandrKedrov** [reply]
I will investigate it when I have spare time. Thanks for the directions!

**Reply by AlexandrKedrov** [reply]
> I can't reproduce this. Not with the RTS sample, not with your project. Tried a Debug and a Dev build. I've also not encountered this issue in general. The input problem that I do know about, was specific to TypeScript and is gone since we switched to AngelScript.
> 
> This appears to be specific to your machine. Maybe some anit-virus software that's blocking access to certain files?
> 
> I think you'll have to debug this issue. Check the logs for any errors. Put a breakpoint in `ezAngelScriptResource::UpdateContent` to see whether any resource fails to load. The RML UI stuff seems to load for you at least partially, not sure where to check what's going wrong there, though it should show up in the log.
> 
> Build the engine in a Debug build, and use the plugin that's mentioned here (https://ezengine.net/pages/docs/debugging/debug-cpp.html) to attach to the child processes.
> 
> Sorry, I don't think I can help you more than that directly, but let me know if you have further questions.

Hi!
I've solved my problem
For AS I had to manually copy `AngelScript.dll`
For RML UI I had to manually copy `RmlUI\rmlui-default\assets.png`

Is it because these files are excluded by `CommonData.ezExportFilter` and `CommonBinaries.ezExportFilter`?

**Comment by jankrassnigg** [comment]
I somehow missed that you mentioned exporting the project. Yes, when doing a full project export, it was indeed missing the AngelScript.dll and an image file for RmlUI. If fixed both now. However, for RmlUI this is only a local fix for the RTS project (see https://github.com/ezEngine/ezEngine/commit/a64012c95add597a7a22e560969beddfdce894f7). This would need to be done in other projects, as well, if you use the sample data. For a proper fix, #1971 needs to be implemented.

Thanks for reporting both the problems and a solution, that helped me figure out the actual reason.
